superset/assets/src/dashboard/reducers/dashboardLayout.js

```diff
@@ -255,6 +256,29 @@ const actionHandlers = {
 
     return nextEntities;
   },
+
+  [UPDATE_COMPONENTS_PARENTS_LIST](state) {
+    const nextState = {
+      ...state,
+    };
+
+    const doUpdate = currentComponent => {
```

Comment:
can you please aadd a component for what this is doing? since it's somewhat complex and very important!

Reply:
I separated this function out into a util function, so that easier to add test.